float32 q
float32 q_dot